exonum/src/blockchain/mod.rs

```diff
@@ -275,6 +275,13 @@ impl Blockchain {
                 schema.tx_location_by_tx_hash_mut().put(hash, location);
             }
 
+            // Invoke handle_execute for each service
+            fork.checkpoint();
+            for service in self.service_map.values() {
+                service.handle_execute(&mut fork);
```

Comment:
This calls into user code, which can panic!

Reply:
Hm, interesting, the same problem happens with `handle_commit` as well...

What actually is the expected behavior here? Seems unfortunate if the bug in a single service (perhaps even specific to a particular transaction) brings down the whole exonum...

Reply:
We need (very) carefully revise guaranties that we can and want to provide.

There are many places where panic can cause problems: `Transaction::verify` and almost all `Service` methods.